### PR TITLE
Move windows package signing to use ESRP

### PR DESCRIPTION
**Pull request by TravisEz13**
# PR Summary

Move windows package signing to use ESRP
  - Use hosted windows pool

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: tools/releaseBuild/azureDevOps/releaseBuild.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ resources:
     type: github
     endpoint: ComplianceGHRepo
     name: PowerShell/compliance
+    ref: create-dest-dir
```
Comment (TravisEz13) [inline tools/releaseBuild/azureDevOps/releaseBuild.yml:19]:
```suggestion
    ref: master
```

File: tools/releaseBuild/azureDevOps/releaseBuild.yml [hunk 2]
```diff
@@ -16,6 +16,7 @@ resources:
     type: github
     endpoint: ComplianceGHRepo
     name: PowerShell/compliance
+    ref: master
```
Comment (adityapatwardhan) [inline tools/releaseBuild/azureDevOps/releaseBuild.yml:19]: I believe this is the default and it needs to be in `ref/heads/master` format
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/resources?view=azure-devops&tabs=schema#resources-repositories

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: tools/releaseBuild/azureDevOps/releaseBuild.yml (hunk 2 above)
Comment (TravisEz13) [reply tools/releaseBuild/azureDevOps/releaseBuild.yml:19]: branch name works and I'm contantly having to add and remove this.  I'd like to keep it for easier testing, and returning it back to it's default.

Comment (TravisEz13) [reply tools/releaseBuild/azureDevOps/releaseBuild.yml:19]: See commit:  [`cc463a6` (#14060)](https://github.com/PowerShell/PowerShell/pull/14060/commits/cc463a61e14c88a0b837a548d38748f392873d15)

**Review by rjmholt — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**